src/lib/provable/merkle-list.ts

```diff
@@ -191,6 +191,27 @@ class MerkleList<T> implements MerkleListBase<T> {
     return new this.Constructor({ hash: this.hash, data });
   }
 
+  /**
+   * Iterate through the list in a fixed number of steps any apply a given callback on each element.
+   *
+   * Proves that the iteration traverses the entire list.
+   * Once past the last element, dummy elements will be passed to the callback.
+   *
+   * Note: There are no guarantees about the contents of dummy elements, so the callback is expected
+   * to handle the `isDummy` flag separately.
+   */
+  forEach(
+    length: number,
+    callback: (element: T, isDummy: Bool, i: number) => void
```

Comment:
API for iterating a MerkleList without having to explicitly instantiate an iterator

this is safer than `.next()` since it does `assertAtEnd()` for you